src/services/completions.ts

```diff
@@ -5472,9 +5472,10 @@ function getJsDocTagAtPosition(node: Node, position: number): JSDocTag | undefin
 /** @internal */
 export function getPropertiesForObjectExpression(contextualType: Type, completionsType: Type | undefined, obj: ObjectLiteralExpression | JsxAttributes, checker: TypeChecker): Symbol[] {
     const hasCompletionsType = completionsType && completionsType !== contextualType;
+    const promiseFilteredContextualType = checker.getUnionType(filter(contextualType.flags & TypeFlags.Union ? (contextualType as UnionType).types : [contextualType], t => !checker.getAwaitedType(t) || !checker.getPromisedTypeOfPromise(t)));

```

Comment:
Why are both `getAwaitedType` and `getPromisedTypeOfPromise` needed?

Reply:
Looking at it again, I don't think I need the `getAwaitedType`. I deleted it. Thank you.